docs/compiler-options.md

```diff
@@ -12,6 +12,7 @@
 | `-Oinline` `-Ono-inline`                                                 | Disabled      | [inline](#inline)                         |
 | `-Ocheck-net-size` `-Ono-check-net-size`                                 | Disabled      | [check-net-size](#check-net-size)         |
 | `-Oadt-scott` `-Oadt-num-scott`                                          | adt-num-scott | [adt-encoding](#adt-encoding)             |
+| `-Otype-check` `-Ono-type-check`                                         | type-check    |  Checks the type compatibility            |
```

Comment:
This is not what the type checker does. Also, please do like the others and point to the section